aws/sdk/examples/lambda-invoke-function/src/main.rs

```diff
@@ -2,52 +2,139 @@
+        // returns a response object containing the following:
+        //
+        // StatusCode: The HTTP status code is in the 200 range for a
+        //   successful request
+        //
+        // ExecutedVersion: The version of the function that executed. When
+        //   you invoke a function with an alias, this indicates which version
+        //   the alias resolved to.
+        //
+        // FunctionError: If present, indicates that an error occurred during
+        //   function execution. Details about the error are included in the
+        //   response payload.
+        //
+        // LogResult: The last 4 KB of the execution log, which is base64
+        //    encoded.
+        //
+        // Payload: The response from the function, or an error object.
+        //
+        // For our example, we are just going to interact with the payload
+        // portion of the response
         Ok(resp) => {
-            println!("Response:");
-            println!("  {:?}", resp.payload);
+            match resp.payload {
+                // I'm sure there is some better idiomatic Rust way to do this
+                // but here we are. The payload response is a blob of binary
+                // data that we need to turn into something useful.
+                Some(blob) => {
+                    // I assign the contents of the blob to another variable
+                    // so that I can borrow it later without the compiler being
+                    // mad at me.
+                    let random_inner_crap = blob.into_inner();
+                    let s = str::from_utf8(&random_inner_crap).unwrap();
+                    println!("Response:");
+                    println!("  {:?}", s);
+                }
+                // I don't think a Lambda function can NOT return a payload
+                // without instead returning an error, so this *should* be
+                // redundant.
+                _ => ()
+            }
+            
         }
+        // If we want to handle a specific error in a specific way, we can
+        // match it here. The SDK will automatically attempt to rety errors
+        // that are considered 'retryable'. These types of errors generally
+        // include responses where a service would return an HTTP 500 code to 
+        // indicate a server-side error. This means that the errors received by
+        // your code here only need to address the rest of the errors that the
+        // SDK wouldn't know how to 'retry', such as when the resource doesn't
+        // exist.
+        //
+        // For our example, we will simply print that the function doesn't
+        // exist and return a non-zero exit code to indicate a failure.
         Err(SdkError::ServiceError { err, .. })
             if matches!(err.kind, InvokeErrorKind::ResourceNotFoundError(_)) =>
         {
             println!("This lambda function does not exist");
```

Comment:
there's actually some more info in these errors (especially after #374 was merged). I'd recommend printing err:
```suggestion
            println!("This lambda function does not exist: {}", err);
```

Reply:
Do we need the error message? since we caught the "ResourceNotFound" Exception and we told it what resource to look for, we know what the error message is.

Reply:
nvm, I made the change and updated the other error arm so they're consistently just a single `println` statement